src/examples/colors/color_grammar.py

```diff
@@ -9,61 +10,65 @@ def __getstate__(self):
         # or cause other conflicts
         state = self.__dict__.copy()
         # Print unpicklable entries
-        if('f' in state):
+        if "f" in state:
             print(f"Unpickled: {state['f']}")
             # remove unpicklable entries
-            del state['f']
+            del state["f"]
+
     def __setstate__(self, state):
         """Used for deserializing instances"""
         # restore instance attributes
         self.__dict__.update(state)
+
     def __hash__(self):
-        if(not hasattr(self, '_dist') or self._dist is None):
+        if not hasattr(self, "_dist") or self._dist is None:
             return hash(tuple(self.referents))
         return hash(tuple(self._dist.values()))
+
     def __str__(self):
         return f"Referents:\n\t{','.join(str(referent.name) for referent in self.referents)}\
             \nDistribution:\n\t{self.dist}\n"
 
 
-
 class ColorLanguage(Language):
-    """Language representing a set of color probabilities. 
+    """Language representing a set of color probabilities.
 
     Args:
         Language (_type_): _description_
     """
-    def __init__(self, expressions: tuple[Expression, ...], natural:bool = True, **kwargs):
+
+    def __init__(
+        self, expressions: tuple[Expression, ...], natural: bool = True, **kwargs
+    ):
         super().__init__(expressions, **kwargs)
         self.natural = natural
 
-    natural:bool = True
+    natural: bool = True
```

Comment:
I temporarily set this value because I was having trouble creating a subclass, as `IsNaturalLanguage()` raises a `NotImplementedError` at the moment. It's misplaced, though, I'll move it to the top of the class.